cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -19,41 +20,79 @@ type Manager struct {
 	console     input.Console
 }
 
+// Infrastructure deployment options
+type DeployOptions struct {
+	// Whether or not the session supports interactive user input
+	Interactive bool
+}
+
+// Infrastructure destroy options
+type DestroyOptions struct {
+	// Whether or not the session supports interactive user input
+	Interactive bool
+	// Whether or not to force the deletion of resources without prompting the user
+	Force bool
+	// Whether or not to purge any key vaults associated with the deployment
+	Purge bool
+}
+
 // Prepares for an infrastructure provision operation
 func (m *Manager) Preview(ctx context.Context, interactive bool) (*PreviewResult, error) {
 	previewResult, err := m.preview(ctx, interactive)
 	if err != nil {
 		return nil, err
 	}
 
-	updated, err := m.ensureParameters(ctx, &previewResult.Preview)
-	if err != nil {
-		return nil, err
-	}
+	return previewResult, nil
+}
 
-	if updated {
-		if err := m.provider.UpdatePlan(ctx, previewResult.Preview); err != nil {
-			return nil, fmt.Errorf("updating deployment parameters: %w", err)
-		}
+// Gets the latest deployment details for the specified scope
+func (m *Manager) GetDeployment(ctx context.Context, scope Scope, interactive bool) (*DeployResult, error) {
+	var deployResult *DeployResult
+
+	spinner := spin.NewSpinner("Retrieving Azure Deployment")
+	err := spinner.Run(func() error {
+		queryTask := m.provider.GetDeployment(ctx, scope)
+
+		go func() {
+			for progressReport := range queryTask.Progress() {
+				if interactive {
```

Comment:
Should we launch this whole goroutine only if interactive == true?

Reply:
We'll let the routine run for now, just not reporting any updates to the console.